### PR TITLE
MM-11084 & MM-11100: More addaptative permissions tree

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Changed a bit how permissions_tree is built, allowing a initial phase
for add/remove/change things. Basically now it adds certain permissions
to the list of permissions based on the configuration (if you have
Emojis, Slash commands, webhooks or OAuth enabled).

Besides change the id (to affect the title/description) of the "Team"
permissions group when you are in the team scope, because there the Team
creation makes no sense.

#### Ticket Link
[MM-11084](https://mattermost.atlassian.net/browse/MM-11084)
[MM-11100](https://mattermost.atlassian.net/browse/MM-11100)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

File: components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            editTimeLimitModalIsVisible: false,
+        };
+
+        this.ADDITIONAL_VALUES = {
+            edit_post: {
+                editTimeLimitButton: <EditPostTimeLimitButton onClick={() => this.onClickEditPostTimeLimitButton()}/>,
```
Comment (hmhealey) [inline components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx]: Why use an arrow function here?

File: components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                        role={this.props.role}
+                        parentRole={this.props.parentRole}
+                        scope={this.props.scope}
+                        combined={false}
+                        onChange={this.toggleGroup}
+                        root={true}
+                    />
+                </div>
+                <EditPostTimeLimitModal
+                    onClose={() => this.setState({editTimeLimitModalIsVisible: false})}
```
Comment (hmhealey) [inline components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx]: Similarly, this should be made into a method of the PermissionsTree or else any time this re-renders, the EditPostTimeLimitModal will as well

File: components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx (hunk 1 above)
Comment (jespino) [reply components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx]: Actually, It wasn't part of my change. I can change it to use directly this.onClickEditPostTimeLimitButton. @mkraft any concern about it?

File: components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx (hunk 2 above)
Comment (jespino) [reply components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx]: It wasn't part of my change, but I can move that into a method.

File: components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx (hunk 1 above)
Comment (jespino) [reply components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx]: Done

File: components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx (hunk 2 above)
Comment (jespino) [reply components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx]: Done

**Comment by jespino** [comment]
@hmhealey PTAL

**Pull request merged**